test_communication/CMakeLists.txt

```diff
@@ -449,6 +454,11 @@ if(BUILD_TESTING)
   custom_executable(test_action_server_cpp
     "test/test_action_server.cpp")
 
+  # Regression test for https://github.com/ros2/rmw_fastrtps/issues/715
+  custom_executable(test_nested_types
+    "test/test_nested_types.cpp")
+  target_link_libraries(test_nested_types rcpputils::rcpputils)
```

Comment:
although we did not see this issue with other Tier I RMW implementations, they still need to be tested here? I think all RMW implementations need to be tested, just checking.

Reply:
This does test it for the other RMWs, right?
This is testing for expected, and potentially unintuitive behavior breaks anyway, so it's good to test 👍

Reply:
> This does test it for the other RMWs, right?

Correct.  Because this test is listed in the `targets` macro down below, and then the `targets` macro is used in `call_for_each_rmw_implementation` even further below, it will be run with every RMW implementation.  This is done by setting the `RMW_IMPLEMENTATION` environment variable at runtime.